src/components/Tables/TableFilterBar.tsx

```diff
@@ -1,70 +1,100 @@
+    shownHeaderKeys: readonly (keyof Data)[];
 }
 
-const StyledFilterAccordion = styled(FilterAccordion)`
-    width: 100%;
-    overflow: visible;
+const StyledButton = styled(Button)`
+    align-self: center;
+    justify-self: flex-end;
 `;
 
-const StyledFilterBar = styled.input`
-    height: 39px;
-    font-size: 14px;
-    width: 15rem;
+const Styling = styled.div`
+    background-color: transparent;
+    display: flex;
+    align-items: stretch;
+    flex-wrap: wrap;
+    padding: 0.5rem 0;
+    gap: 2rem;
     overflow: visible;
-    box-shadow: none;
-    padding: 4px 12px 4px 12px;
+    display: flex;
+    width: 100%;
 
-    &:focus {
-        outline: none;
+    @media (min-width: 500px) {
+        flex-wrap: nowrap;
     }
 `;
 
-interface Props {
-    filterText: FilterText;
-    onFilter: (event: React.ChangeEvent<HTMLInputElement>, filterField: string) => void;
-    handleClear: () => void;
-    headers: [string, string][];
-    filterKeys: string[];
-    toggleableHeaders?: string[];
-    setShownHeaderKeys: (headers: string[]) => void;
-    shownHeaderKeys: string[];
-}
+const FilterHeading = styled.h2`
+    width: 100%;
+    text-align: left;
+    font-size: large;
+`;
+
+const Grow = styled.div`
+    flex-grow: 1;
+`;
+
+const TableFilterBar = <Data extends unknown>(props: Props<Data>): React.ReactElement => {
+    if (props.filters.length === 0 && props.toggleableHeaders.length === 0) {
+        return <></>;
+    }
 
-const TableFilterBar: React.FC<Props> = (props) => {
     return (
         <>
-            <Button variant="contained" onClick={props.handleClear}>
-                Clear
-            </Button>
-            {props.headers
-                .filter(([key]) => props.filterKeys?.includes(key) ?? true)
-                .map(([key, value]) => {
-                    return (
-                        <StyledFilterBar
-                            key={key}
-                            type="text"
-                            value={props.filterText[key] ?? ""}
-                            placeholder={`Filter by ${value}`}
-                            onChange={(event: React.ChangeEvent<HTMLInputElement>) =>
-                                props.onFilter(event, key)
-                            }
-                        />
-                    );
-                })}
-            <StyledFilterAccordion
-                toggleableHeaders={props.toggleableHeaders}
-                shownHeaderKeys={props.shownHeaderKeys}
-                setShownHeaderKeys={props.setShownHeaderKeys}
-                headers={props.headers}
-            />
+            <FilterHeading>Filters</FilterHeading>
+            <Styling>
+                {props.filters.length > 0 && (
+                    <>
+                        {props.filters.map((filter, index) => {
+                            const onFilter = (state: any): void => {
+                                const newFilters = [...props.filters];
```

Comment:
you've used this 'set state of one index of a list' a lot - can we extract to a function, or just give components a setter function that 'does this for you'?